Rihards Macijauskis

RihardsMacijauskis opened the pull request:
Couldn't set up PHPUnit for unit testing on my computer yet due to technical difficulties, but it will be done by tomorrow.
So here's the code for the main program.

RihardsMacijauskis committed:
Add files via upload

Couldn't set up PHPUnit for unit testing on my computer yet due to technical difficulties, but it will be done by tomorrow.
So here's the code for the main program.

RihardsMacijauskis committed:
Update fooBar.php

RihardsMacijauskis committed:
Add files via upload

RihardsMacijauskis committed:
Update fooBar.php

Got rid of some bugs caused by unit testing

RihardsMacijauskis committed:
Update fooBarTest.php

Corrected spacing in lines 25, 30 , 35 (words in quotation marks).

OskarsOzols commented:
Please fix it based on comments and my suggestion for pr #16, as there are almost the same problems.

After that, request for review for Your pr, so I will know I need to look at it.

RihardsMacijauskis committed:
Update fooBar.php

Converting "number" to string if it isn't a multiple of 3 or 5.

RihardsMacijauskis committed:
Update fooBarTest.php

Added a integer to string conversion test case.

RihardsMacijauskis reviewed (commented):
Edited some bits, but in general, I don't see any issues with the code, so maybe you could tell a little bit detailed about what's wrong with it (I checked the PSR guidelines, optimized the code, all tests passed etc.).

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,34 @@
+<?php
+class fooBar
```
Classname case usage not according to PSR.

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,34 @@
+    function checkFooBar($number)
+    {
+        $result = "";
+        //Check if a number is a positive integer
```
Please, remove all comments.

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,34 @@
+            //Check if a number is a multiple of 3
+            if (($number % 3) == 0)
+            {
+                $result = "Foo ";
```
Wrong value

OskarsOzols commented on fooBarTest.php:
```diff
@@ -0,0 +1,45 @@
+     public function testIntToStr()
+    {
+        $number = new FooBar();
+        $this->assertEquals("string", gettype($number->checkFooBar(7)));
```
This test should fail

RihardsMacijauskis replied:
It doesn't, the "gettype" returns value "string"(because of the strval() function) while i'm comparing it with the value "string".

OskarsOzols replied:
Ok, I understood. Better change or add check for specific result checkFooBar will return on '7'.

OskarsOzols commented on fooBarTest.php:
```diff
@@ -0,0 +1,45 @@
+{
+    public function testFloat()
+    {
+        $number = new fooBar;
```
Wrong class name

OskarsOzols commented on fooBarTest.php:
```diff
@@ -0,0 +1,45 @@
+    public function testFoo()
+    {
+        $number = new fooBar;
+        $this->assertEquals($number->checkFooBar(3) , "Foo ");
```
Wrong value, shouldn't be space included.

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,34 @@
+        }
+        else
+        {
+            $result = "Number is not a positive integer";
```
Should throw exception.

RihardsMacijauskis committed:
Update fooBar.php

Changed class name so it complies with PSR guidelines, got rid of comments, whitespaces, edited the else statement so it throws an exception.

RihardsMacijauskis committed:
Update fooBarTest.php

Edited the class names, added the exception tests, deleted whitespaces.

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,30 @@
+        }
+        else
+        {
+            throw new Exception();
```
This exception is too general, You need to find one which relates to input arguments.

OskarsOzols commented on fooBar.php:
```diff
@@ -0,0 +1,30 @@
+<?php
```
Please add directive to use strict_types

OskarsOzols commented on fooBarTest.php:
```diff
@@ -0,0 +1,48 @@
+{
+     public function testFloatException()
+    {
+        $number = new FooBar;
```
I'm not sure, but shouldn't there be () after class name?
